includes/class-wc-stripe-helper.php

```diff
@@ -373,7 +375,20 @@ public static function get_webhook_url() {
 	public static function get_order_by_source_id( $source_id ) {
 		global $wpdb;
 
-		$order_id = $wpdb->get_var( $wpdb->prepare( "SELECT DISTINCT ID FROM $wpdb->posts as posts LEFT JOIN $wpdb->postmeta as meta ON posts.ID = meta.post_id WHERE meta.meta_value = %s AND meta.meta_key = %s", $source_id, '_stripe_source_id' ) );
+		if ( class_exists( 'OrderUtil' ) && OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			$orders   = wc_get_orders(
```

Comment:
Worth mentioning this works for both COT and WP post tables. I wasn't sure if we should use `wc_get_orders` for both though. Maybe in a few months when COT has been highly tested we can remove the query and use `wc_get_orders` for both scenarios.

Reply:
Had a brief chat with @RadoslavGeorgiev about this since I thought he introduced the SQL query on one of his [PR](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/826/files?file-filters%5B%5D=.php&show-viewed-files=true#diff-dbc2cfbfe4b247c27c20d86922819977adc6f7723c0b047cb9722a2f47b7ee73R440).

Here's his response when I asked why a SQL query is used instead of `wc_get_orders`.

>It was done for performance issues. The meta table on WP sites can be absolutely abused by a ton of extensions, which is why they switched to this query.

(For example, here's a PR where the SQL query was optimized even more: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/637 )

I completely agree. The purpose of COT is to have a separate table to optimize the data lookup for WooCommerce, and I think there's no reason to use a plain SQL query here. `wc_get_orders` should be enough.

Reply:
I agree with how you've approached this and that we shouldn't be messing with the non-COT query 👍